skrub/datasets/tests/test_utils.py

```diff
@@ -37,18 +38,45 @@ def test_get_data_home_str():
         assert data_home.exists()
 
 
-def test_get_data_home_None():
+def test_get_data_home_path():
     """
-    Test function for ``get_data_home()`` with `data_home` set to `None`.
+    Test function for ``get_data_home()`` when `data_home` is a pathlib.Path.
     """
-    # get path of the folder 'skrub_data' in the user home folder
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmpdirpath = Path(tmpdirname)
+        # get_data_home will point to a pre-existing folder
+        data_home = get_data_home(data_home=tmpdirname)
+        assert data_home == tmpdirpath
+        assert data_home.exists()
+
+        # if the folder is missing it will be created
+        shutil.rmtree(data_home)
+        assert not data_home.exists()
+        assert not tmpdirpath.exists()
+        data_home = get_data_home(data_home=tmpdirname)
+        assert data_home.exists()
+
+
+def test_get_data_home_default():
+    """Test function for ``get_data_home()`` with default `data_home`."""
+    # We should take care of not deleting the folder if our user
+    # already cached some data
     user_path = Path("~") / "skrub_data"
+    is_already_existing = user_path.exists()
+
+    data_home = get_data_home(data_home=None)
+    assert data_home == user_path
+    assert data_home.exists()
```

Comment:
This is too tightly coupled with the implementation.
A better way to test that would be for example to assert there is a directory which contains the name `skrub` in the home folder.